custom-templates/priv_instances_public_nlb_ssl/neo4j-nlb-priv-ssl.template.yaml

```diff
@@ -0,0 +1,857 @@
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref Neo4jPublicRouteTable
+      SubnetId: !Ref Neo4jPublicSubnet2
+
+  #For Public Subnet 3
+  Neo4jPublicSubnetRouteTableAssociation3:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref Neo4jPublicRouteTable
+      SubnetId: !Ref Neo4jPublicSubnet3
+
+  #For Private Subnet
+  Neo4jPrivateSubnetRouteTableAssociation1:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref Neo4jPrivateRouteTable
+      SubnetId: !Ref Neo4jPrivateSubnet1
+
+  #For Private Subnets
+  Neo4jPrivateSubnetRouteTableAssociation2:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref Neo4jPrivateRouteTable
+      SubnetId: !Ref Neo4jPrivateSubnet2
+
+  #For Private Subnets
+  Neo4jPrivateSubnetRouteTableAssociation3:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref Neo4jPrivateRouteTable
+      SubnetId: !Ref Neo4jPrivateSubnet3
+
+  #For Private Subnets
+  Neo4jNATGateway:
+    Type: AWS::EC2::NatGateway
+    Properties:
+      AllocationId: !GetAtt Neo4jNATGatewayEIP.AllocationId
+      SubnetId: !Ref Neo4jPublicSubnet1
+      Tags:
+        - Key: StackID
+          Value: !Ref 'AWS::StackId'
+        - Key: Name
+          Value: !Ref 'AWS::StackName'
+
+ #For Private Subnets
+  Neo4jNATGatewayEIP:
+    Type: AWS::EC2::EIP
+    Properties:
+      Domain: vpc
+      Tags:
+        - Key: StackID
+          Value: !Ref 'AWS::StackId'
+        - Key: Name
+          Value: !Ref 'AWS::StackName'
+
+  #VPC/Instance Security Group
+  Neo4jInstanceSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Security group for instances, open across the VPC
+      VpcId: !Ref Neo4jVPC
+      SecurityGroupIngress:
+        - IpProtocol: -1
+          CidrIp: !GetAtt Neo4jVPC.CidrBlock
+        - IpProtocol: tcp
+          FromPort: 7474
+          ToPort: 7474
+          CidrIp: 0.0.0.0/0
+        - IpProtocol: tcp
+          FromPort: 7687
+          ToPort: 7687
+          CidrIp: 0.0.0.0/0
+
+  #Bastion Security Group
+  Neo4jBastionSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Security Group for Bastion Instance
+      VpcId: !Ref Neo4jVPC
+      SecurityGroupIngress:
+      - IpProtocol: tcp
+        FromPort: 22
+        ToPort: 22
+        CidrIp: !Ref BastionSSHCIDR
+
+  Neo4jAutoScalingGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    DependsOn: Neo4jNATGateway
+    CreationPolicy:
+      ResourceSignal:
+        Count: !Ref NumberOfServers
+        Timeout: PT8M
+    Properties:
+      AvailabilityZones: !If [CreateCluster, [!Select [ 0, Fn::GetAZs: !Ref 'AWS::Region' ] , !Select [ 1, Fn::GetAZs: !Ref 'AWS::Region' ] , !Select [ 2, Fn::GetAZs: !Ref 'AWS::Region' ]],[!Select [ 0, Fn::GetAZs: !Ref 'AWS::Region' ]]]
```

Comment:
Given this is deploying to a minimum 3 `NumberOfServers` and given all 3 subnets are always created, this line can just statically define `[!Select [ 0, Fn::GetAZs: !Ref 'AWS::Region' ] , !Select [ 1, Fn::GetAZs: !Ref 'AWS::Region' ] , !Select [ 2, Fn::GetAZs: !Ref 'AWS::Region' ]`